test/test_irods_http_api.py

```diff
@@ -206,23 +210,51 @@ def test_common_operations(self):
         # Remove the collection.
         data = {'op': 'remove', 'lpath': new_collection_path}
         r = requests.post(self.url_endpoint, headers=headers, data=data)
-        #print(r.content) # Debug
+        logging.debug(r.content)
         self.assertEqual(r.status_code, 200)
 
         # Stat the collection to show that it does not exist.
         params = {'op': 'stat', 'lpath': new_collection_path}
         r = requests.get(self.url_endpoint, headers=headers, params=params)
-        #print(r.content) # Debug
+        logging.debug(r.content)
         self.assertEqual(r.status_code, 400)
 
         stat_info = r.json()
-        self.assertEqual(stat_info['irods_response']['status_code'], -170000) # NOT_A_COLLECTION
+        self.assertEqual(stat_info['irods_response']['status_code'], irods_error_codes.NOT_A_COLLECTION)
+
+    def test_creating_a_collection_with_insufficient_permissions_results_in_an_error(self):
+        rodsuser_headers = {'Authorization': 'Bearer ' + self.rodsuser_bearer_token}
+        collection = f'/{self.zone_name}/home/{self.rodsadmin_username}/not_allowed'
+
+        # Attempting to create a collection with insufficient permissions and the
+        # "create-intermediates" parameter not set to 1 results in an error.
+        r = requests.post(self.url_endpoint, headers=rodsuser_headers, data={
+            'op': 'create',
+            'lpath': collection
+        })
+        logging.debug(r.content)
+        self.assertEqual(r.status_code, 200)
+        self.assertEqual(r.json()['irods_response']['status_code'], irods_error_codes.OBJ_PATH_DOES_NOT_EXIST)
+
+        # Attempting to create a collection with insufficient permissions and the
+        # "create-intermediates" parameter set to 1 results in an error.
+        r = requests.post(self.url_endpoint, headers=rodsuser_headers, data={
+            'op': 'create',
+            'lpath': f'{collection}/more/path/elements', # Guards against incorrect implementations.
+            'create-intermediates': 1
+        })
+        logging.debug(r.content)
+        self.assertEqual(r.status_code, 200)
+        # Sadly, there's nothing we can do within the HTTP API implementation to make
+        # this call and the one before result in the same iRODS error code. Fixing this
+        # requires a change in the iRODS server.
```

Comment:
Has an issue already been filed for this?

Reply:
Not yet. It's on the TODO list though.

Reply:
Wait, I may have gotten this comment mixed up with some other logic.

This is about use of the recursive flag resulting in a different error code as compared to when the recursive flag isn't used.

Will confirm shortly.

Reply:
Created https://github.com/irods/irods/issues/7461 for this.

Noting here that the `OBJ_PATH_DOES_NOT_EXIST` error is due to the filesystem library. The `rxCollCreate` error codes captured in the irods/irods issue are the API specific error codes.